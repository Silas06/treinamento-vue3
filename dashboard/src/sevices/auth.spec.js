import Authservice from './auth'
import mockAxios from 'axios'

jest.mock('axios')

describe('teste auth', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('deve retornar um token quando fizer login', async () => {
    const token = '123.123.122'
    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: { token } })
    })

    const response = await Authservice(mockAxios).login({
      email: 'igor@igor.me',
      password: '123'
    })

    expect(response.data).toHaveProperty('token')
    expect(response).toMatchSnapshot()
  })

  it('retorna o usuario que esta registrado', async () => {
    const user = {
      name: 'Igor',
      password: '123',
      email: 'igor@igor.me'
    }

    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: user })
    })

    const response = await Authservice(mockAxios).register(user)
    expect(response.data).toHaveProperty('name')
    expect(response.data).toHaveProperty('password')
    expect(response.data).toHaveProperty('email')
    expect(response).toMatchSnapshot()
  })

  it('Erro na request', async () => {
    const errors = { status: 404, statusText: 'Not Found' }
    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ request: errors })
    })

    const response = await Authservice(mockAxios).login({
      email: 'igor@igor.me',
      password: '123'
    })
    expect(response.errors).toHaveProperty('status')
    expect(response.errors).toHaveProperty('statusText')
  })
})
