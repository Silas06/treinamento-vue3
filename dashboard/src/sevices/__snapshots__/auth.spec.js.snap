// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`teste auth deve retornar um token quando fizer login 1`] = `
Object {
  "data": Object {
    "token": "123.123.122",
  },
  "errors": null,
}
`;

exports[`teste auth retorna o usuario que esta registrado 1`] = `
Object {
  "data": Object {
    "email": "igor@igor.me",
    "name": "Igor",
    "password": "123",
  },
  "errors": null,
}
`;
